custom_code/views.py

```diff
@@ -58,6 +61,32 @@ def get_context_data(self, **kwargs):
         context['extra_form'] = TargetListExtraFormset()
         return context
 
+class TargetDetailView(Raise403PermissionRequiredMixin, DetailView):
+    """
+    View that handles the display of the target details. Requires authorization.
+    """
+    permission_required = 'tom_targets.view_target'
+    model = Target
+    template_name = 'tom_targets/target_detail.html'
+
+    def get_context_data(self, *args, **kwargs):
+        """
+        Adds the ``DataProductUploadForm`` to the context and prepopulates the hidden fields.
+
+        :returns: context object
+        :rtype: dict
+        """
+        context = super().get_context_data(*args, **kwargs)
+        observation_template_form = ApplyObservationTemplateForm(initial={'target': self.get_object()})
+        if any(self.request.GET.get(x) for x in ['observation_template', 'cadence_strategy', 'cadence_frequency']):
+            initial = {'target': self.object}
+            initial.update(self.request.GET)
+            observation_template_form = ApplyObservationTemplateForm(
+                initial=initial
+            )
+        observation_template_form.fields['target'].widget = HiddenInput()
+        context['observation_template_form'] = observation_template_form
+        return context
```

Comment:
I think this is identical to the built-in `TargetDetailView`. If so, why do you need it?